drivers: can: set default y for stm32 CAN driver

alexanderwachter opened the pull request:
Set CAN_STM32 default to y.
This makes sense because it anyway depends on SOC_FAMILY_STM32.

Commit:
drivers: can: set default y for stm32 CAN driver

Set CAN_STM32 default to y. This makes sense because it anyway depends
on SOC_FAMILY_STM32.

Signed-off-by: Alexander Wachter <alexander.wachter@student.tugraz.at>

carlescufi commented on drivers/can/Kconfig.stm32:
```diff
@@ -9,6 +9,7 @@
 config CAN_STM32
 	bool "STM32 CAN Driver"
 	depends on SOC_FAMILY_STM32
+	default y
```
typically we do this per-board instead. Like for example: 
https://github.com/zephyrproject-rtos/zephyr/blob/master/boards/arm/stm32f4_disco/Kconfig.defconfig#L22

alexanderwachter replied:
This only selects the stm32 CAN driver if CAN is enabled in general. It's annoying to select CAN and the select the hardware driver. For example if you want to run the API tests it's helpful.

carlescufi replied:
Sure, but what about SoCs in the `SOC_FAMILY_STM32` that don't support CAN at all?

alexanderwachter replied:
This series won't set CONFIG_CAN to y

ydamigos replied:
I believe this is already done in `soc/arm/st_stm32/common/Kconfig.defconfig.series`, isn't it?

alexanderwachter replied:
@ydamigos you are right!

erwango reviewed (commented):
This makes indeed sense, as there is no alternative.
This is not directly related, but can you edit the help section and documents the list of series on which CAN is known to be working?

alexanderwachter commented:
@erwango This list is continuously growing. So far all stm32 series with CAN controller on board worked.

alexanderwachter commented:
@carlescufi @ Hopefully we can get rid of this Kconfig options and use dts status="ok" for that soon.

erwango commented:
> This list is continuously growing. So far all stm32 series with CAN controller on board worked.

(Great)
Anyway we could consider the help section message is out of date and deserves an update.

Commit:
drivers: can: remove tested on stm32F0 series statement

Remove statement that it's tested on F0 series because it's tested
on F0, F7 and L4 now and no issues occured.

Signed-off-by: Alexander Wachter <alexander.wachter@student.tugraz.at>

alexanderwachter commented:
@erwango Now I know what you mean. I totally forgot about that statement. It's removed now

ydamigos commented on drivers/can/Kconfig.stm32:
```diff
@@ -9,9 +9,10 @@
 	select USE_STM32_HAL_CAN
 	help
-	  Enable STM32 CAN Driver (tested on stm32F0 series)
+	  Enable STM32 CAN Driver
```
@alexanderwachter Why did you close the PR? We could use it to update help section.

alexanderwachter replied:
I'll open a new one.
